pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -2227,6 +2230,20 @@ public ResultSet getTypeInfo() throws SQLException {
     rs.close();
     stmt.close();
 
+    // Per JDBC API, we should sort result by SqlType value
+    Collections.sort(v, new Comparator<byte[][]>() {
+      @Override
+      public int compare(byte[][] o1, byte[][] o2) {
+        try {
+          int i1 = Integer.parseInt(connection.getEncoding().decode(o1[1]));
```

Comment:
Would you please perform `parseInt/decode` stuff outside of Comparator to avoid repeated decode/parse in the sort loop?

Reply:
Hi @vlsi, The decode part is needed in the loop because we need the byte[1] "sqlType" as input byte. Moving it outside means I have to create new list of byte array and associate the origingal sqlType int value as one element. And again to remove the int sqlType as final result after the sorting for return value. Can you please confirm is that what you like to see?